From 78a81e53842fab0138a9549af663a8f91f8d63c9 Mon Sep 17 00:00:00 2001
From: jmpfbmx <jmpf.bmx@gmail.com>
Date: Mon, 31 Jul 2021 11:32:17 +0100
Subject: [PATCH] mkbootimg: Don't eliminate meaningless non-zero values of
 second_offset and ramdisk_offset"

* Breaks boots in MTK Devices that needs SaR

This reverts commit 983425fe948027f60562088f270a80abbfb1018c.
Co-authored-by: bengris32 <bengris32@protonmail.ch>
Signed-off-by: jmpfbmx <jmpf.bmx@gmail.com>
---
 mkbootimg.py | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/mkbootimg.py b/mkbootimg.py
index c5ff1e7..e761893 100644
--- a/mkbootimg.py
+++ b/mkbootimg.py
@@ -114,16 +114,14 @@ def write_header(args):
         return write_header_v3(args)
 
     args.output.write(pack('8s', BOOT_MAGIC))
-    final_ramdisk_offset = (args.base + args.ramdisk_offset) if filesize(args.ramdisk) > 0 else 0
-    final_second_offset = (args.base + args.second_offset) if filesize(args.second) > 0 else 0
     args.output.write(pack(
         '10I',
         filesize(args.kernel),                          # size in bytes
         args.base + args.kernel_offset,                 # physical load addr
         filesize(args.ramdisk),                         # size in bytes
-        final_ramdisk_offset,                           # physical load addr
+        args.base + args.ramdisk_offset,                # physical load addr
         filesize(args.second),                          # size in bytes
-        final_second_offset,                            # physical load addr
+        args.base + args.second_offset,                 # physical load addr
         args.base + args.tags_offset,                   # physical addr for kernel tags
         args.pagesize,                                  # flash page size we assume
         max(args.header_version, filesize(args.dt)),    # version of bootimage header or dt size in bytes


